src/main/java/edu/cornell/kfs/gl/businessobject/lookup/CuCurrentAccountBalanceLookupableHelperServiceImpl.java

```diff
@@ -0,0 +1,269 @@
+package edu.cornell.kfs.gl.businessobject.lookup;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.kuali.kfs.coa.businessobject.Account;
+import org.kuali.kfs.gl.businessobject.Balance;
+import org.kuali.kfs.gl.businessobject.CurrentAccountBalance;
+import org.kuali.kfs.gl.businessobject.lookup.CurrentAccountBalanceLookupableHelperServiceImpl;
+import org.kuali.kfs.krad.util.ObjectUtils;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.KFSParameterKeyConstants;
+import org.kuali.kfs.sys.businessobject.SystemOptions;
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+
+import edu.cornell.kfs.sys.CUKFSParameterKeyConstants;
+
+/**
+ * Custom CurrentAccountBalance lookupable helper subclass that adds some enhancements
+ * to the lookup. One is an option for excluding "CB" amounts from certain calculations,
+ * effectively turning the results into Current Fund Balances instead. Another change
+ * is using object type code to perform Current Asset/Liability calculations instead,
+ * and there are associated parameters to go along with that. Some miscellaneous
+ * clean-up and refactoring has been done as well.
+ */
+public class CuCurrentAccountBalanceLookupableHelperServiceImpl extends CurrentAccountBalanceLookupableHelperServiceImpl {
+
+    private static final long serialVersionUID = 2542719296293895780L;
+
+    /**
+     * This override constructs and uses a custom helper class to execute the superclass's processing.
+     * 
+     * @see org.kuali.kfs.gl.businessobject.lookup.CurrentAccountBalanceLookupableHelperServiceImpl#updateCurrentBalance(
+     * org.kuali.kfs.gl.businessobject.CurrentAccountBalance, org.kuali.kfs.gl.businessobject.Balance, java.lang.String)
+     */
+    @Override
+    protected void updateCurrentBalance(CurrentAccountBalance currentBalance, Balance balance, String fiscalPeriod) {
+        new BalanceUpdater(currentBalance, balance, fiscalPeriod)
+                .updateCurrentBalance();
+    }
+
+    /**
+     * Helper class for encapsulating and splitting up the logic
+     * of the updateCurrentBalance() method.

```

Comment:
Are these comments needed?  Maybe rename the class to BalanceUpdaterHelper to reinforce that is just a helper class/